Tests/RoslynPluginGeneratorTests/NuGetPackageHandlerTests.cs

```diff
@@ -17,22 +17,173 @@ public class NuGetPackageHandlerTests
     {
         public TestContext TestContext { get; set; }
 
+        #region Tests
+
         [TestMethod]
         public void NuGet_TestDependencyResolutionFailure()
         {
             // Arrange
             string testDir = TestUtils.CreateTestDirectory(this.TestContext);
             TestLogger logger = new TestLogger();
-            NuGetPackageHandler handler = new NuGetPackageHandler(logger);
+            NuGetPackageHandler handler = new NuGetPackageHandler(testDir, logger);
 
             // Act
             // Fetch a package that should fail due to pre-release dependencies
-            IPackage package = handler.FetchPackage(AnalyzerPluginGenerator.NuGetPackageSource, "codeCracker", null, testDir);
+            IPackage package = handler.FetchPackage("codeCracker", null, testDir);
 
             // Assert
             // No files should have been downloaded
-            Assert.IsTrue(Directory.GetFiles(testDir).Length == 0);
+            Assert.IsTrue(Directory.GetFiles(testDir, "*.*", SearchOption.AllDirectories).Length == 0);
             Assert.IsNull(package);
         }
+
+        [TestMethod]
+        public void FetchPackage_VersionSpecified_CorrectVersionSelected()
```

Comment:
Added unit tests to cover version selection e.g.
* version specified,
* version not specified - > latest released used
* version not specified and no release -> most recent used

The NuGet packages are generated on the fly.